File: fresher_food_backend/FressFood/Models/PasswordResetRequest.cs
namespace FressFood.Models
{
    /// <summary>
    /// Model cho yêu cầu đặt lại mật khẩu
    /// </summary>
    public class PasswordResetRequest
    {
        public string MaYeuCau { get; set; }
        public string Email { get; set; }
        public string? MaNguoiDung { get; set; }
        public string? TenNguoiDung { get; set; }
        public string TrangThai { get; set; } // "Pending", "Approved", "Rejected"
        public DateTime NgayTao { get; set; }
        public DateTime? NgayXuLy { get; set; }
        public string? MaAdminXuLy { get; set; }
        public string? MatKhauMoi { get; set; } // Lưu tạm để gửi email, sau đó xóa
    }

    /// <summary>
    /// Request để tạo yêu cầu đặt lại mật khẩu
    /// </summary>
    public class CreatePasswordResetRequest
    {
        public string Email { get; set; } = string.Empty;
    }

    /// <summary>
    /// Request để admin xử lý yêu cầu
    /// </summary>
    public class ProcessPasswordResetRequest
    {
        public string MaYeuCau { get; set; } = string.Empty;
        public string Action { get; set; } = string.Empty; // "Approve" or "Reject"
        public string? MaAdmin { get; set; }
    }
}

File: fresher_food/lib/models/PasswordResetRequest.dart
class PasswordResetRequest {
  final String maYeuCau;
  final String email;
  final String? maNguoiDung;
  final String? tenNguoiDung;
  final String trangThai; // "Pending", "Approved", "Rejected"
  final DateTime ngayTao;
  final DateTime? ngayXuLy;
  final String? maAdminXuLy;

  PasswordResetRequest({
    required this.maYeuCau,
    required this.email,
    this.maNguoiDung,
    this.tenNguoiDung,
    required this.trangThai,
    required this.ngayTao,
    this.ngayXuLy,
    this.maAdminXuLy,
  });

  factory PasswordResetRequest.fromJson(Map<String, dynamic> json) {
    return PasswordResetRequest(
      maYeuCau: json['maYeuCau']?.toString() ?? '',
      email: json['email']?.toString() ?? '',
      maNguoiDung: json['maNguoiDung']?.toString(),
      tenNguoiDung: json['tenNguoiDung']?.toString(),
      trangThai: json['trangThai']?.toString() ?? 'Pending',
      ngayTao: json['ngayTao'] != null
          ? DateTime.parse(json['ngayTao'].toString())
          : DateTime.now(),
      ngayXuLy: json['ngayXuLy'] != null
          ? DateTime.parse(json['ngayXuLy'].toString())
          : null,
      maAdminXuLy: json['maAdminXuLy']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'maYeuCau': maYeuCau,
      'email': email,
      'maNguoiDung': maNguoiDung,
      'tenNguoiDung': tenNguoiDung,
      'trangThai': trangThai,
      'ngayTao': ngayTao.toIso8601String(),
      'ngayXuLy': ngayXuLy?.toIso8601String(),
      'maAdminXuLy': maAdminXuLy,
    };
  }

  bool get isPending => trangThai == 'Pending';
  bool get isApproved => trangThai == 'Approved';
  bool get isRejected => trangThai == 'Rejected';
}

